src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryReader.cs

```diff
@@ -1414,9 +1394,11 @@ private int ReadArray(DateTime[] array, int offset, int count)
         {
             CheckArray(array, offset, count);
             int actual = Math.Min(count, _arrayCount);
-            for (int i = 0; i < actual; i++)
+            // Try to read in whole array, but don't fail if not possible
+            BufferReader.GetBuffer(actual * ValueHandleLength.DateTime, out _, out _);
```

Comment:
1. Se comment on UnsafeReadarray/ReadRawArrayBytes about chunked reads, if it should be added back then these GetBuffer calls should probably be removed
2. Is it better to use overload with a single out parameter that throws an exception directly if the file ends prematurely ?  The current approach keeps the old behaviour where these "special" types will partially fill arrays in case of premature end-of-file